### DISCUSSION TITLE
errow in setting thermal-conductivity coupled temperature

### DISCUSSION THREAD

**Discussion by DYLDYLDYL** [idaholab/moose#22451] [Q&A Modules: General] [answered]
hi,
I want to set thermal-conductivity coupled temperature.
```
[Materials]
  [thermal]
    type = HeatConductionMaterial
    temp = T
    thermal_conductivity_temperature_function = 0.1*T
  []
[]
```
It shows: The following error occurred in the object "MOOSE Problem", of type "FEProblem".Unable to find function 0.1*T
When I set a parsed function K in [functin], and then thermal_conductivity_temperature_function = k,  it will run.
But parsed function can only couple xyzt.

Thanks in advance for your help!

**Comment by GiudGiud** [comment, marked as answer]
Hello

0.1 * T is not a "Function" in MOOSE because it depends on a variable, and it's not in the format MOOSE expects for two reasons:

- In general only the ParsedXXX objects will accept this sort of syntax (except with constants, constant parsing is widely supported)

- This parameter likely expects a 1D function that maps temperature to thermal conductivity. You can easily reproduce 0.1 T with a real Function by specifying a PiecewiseLinear function for example.

Note: this parameter abuses Functions in that it takes 'x' to be temperature. So instead of having a spatial dependence, they have a temperature dependence. The ParsedFunction you could pass here would be: 0.1 * x

Guillaume

**Reply by DYLDYLDYL** [reply]
Thanks a lot ！Your answer solves my problem perfectly ！
I would like to ask a further question. 
I want to get 2Dor3D function that maps temperature to thermal conductivity,   is PiecewiseLinear function up to the work？

**Reply by GiudGiud** [reply]
we dont support that.
Using T as the 'x' coordinate is kind of a hack.
You definitely could implement something like using P and T as x and y, but it would likely be new code

**Reply by DYLDYLDYL** [reply]
Thanks!

**Reply by DYLDYLDYL** [reply]
Thanks!

**Reply by liao225** [reply]
Hello, excuse me, may I ask if it is correct to use this method?
![11e8b627f965a1ed1fd472fa333b33e](https://github.com/user-attachments/assets/f7007b28-9519-4d33-b055-6af463b41a29)

**Reply by GiudGiud** [reply]
this looks fine to me for a theoretical material

**Reply by GiudGiud** [reply]
actually, no, probably should be 0.1 * t

**Reply by liao225** [reply]
Thanks!
